storage/google/cloud/storage/_signing.py

```diff
@@ -120,8 +122,10 @@ def generate_signed_url(
     response_type=None,
```

Comment:
@tseaver the surface LGTM, though I do have comments on the surface that I'd like your feedback on:
1. This is okay, but I do think the surface can be simplified for `content_md5`, `content_type`, `response_type`, `response_disposition` to be accepted through headers and query parameters.
2. The generate_signed_url() method will act the same, but will instead pass the values for the values in (1) to either header or query parameters.

Reply:
From my reading of the [V2 spec](https://cloud.google.com/storage/docs/access-control/signed-urls#string-components), it mandates placing `Content-MD5` and `Content-Type` differently from headers (they appear before the expiration, and do not have header names).

Reply:
Thanks Tres, I thought it would be cleaner for the underlying design method signatures only (`generate_signed_url_v2` and `generate_signed_url_v4`). I wouldn't expect `generate_signed_url()` to change its method parameters. 

Though, this was mainly a suggestion given v4 doesn't have a similar requirement for `Content-MD5` and `Content-Type`.

Reply:
@tseaver please do move forward with the current design. The prior message was just a suggestion in method signature design.